docs/sharing.md

```diff
@@ -436,11 +436,52 @@ The use of a code management system is important to keep together all the depend
 
 Moreover to guarantee that a pipeline is reproducible it should be self-contained i.e. it should have ideally no dependencies on the hosting environment. By using Nextflow you can achieve this goal following these methods:
 
-### Third party scripts
+### Binary applications
+
+Docker allows you to ship any binary dependencies that you may have in your pipeline to a portable image that is downloaded on-demand and can be executed on any platform where a Docker engine is installed.
+
+In order to use it with Nextflow, create a Docker image containing the tools needed by your pipeline and make it available in the [Docker Hub](https://hub.docker.com).
+
+Then declare in the `nextflow.config` file, that you will include in your project, the name of the Docker image you have created. For example:
+
+```groovy
+process.container = 'my-docker-image'
+docker.enabled = true
+```
+
+In this way when you launch the pipeline execution, the Docker image will be automatically downloaded and used to run your tasks.
+
+Read the {ref}`container-page` page to learn more on how to use containers with Nextflow.
+
+This mix of technologies makes it possible to write self-contained and truly reproducible pipelines which require zero configuration and can be reproduced in any system having a Java VM and a Docker engine installed.
+
+[^id2]: BitBucket provides two types of version control system: Git and Mercurial. Nextflow supports only Git repositories.
+
+### Bundling executables in the workflow
+
+In most cases, software dependencies should be provided by the execution environment ([container](./container.md), [conda](./conda.md)/[spack](./spack.md) environment, or host-native [modules](./process.md#module)). 
+
+In cases where you do not wish to modify the execution environment(s), executable scripts can be included in the `bin` directory in the workflow repository root. This can be useful to make changes that affect task execution across all environments with a single change.
```

Comment:
```suggestion
In cases where you do not wish to modify the execution environment(s), executable scripts can be included in the `bin/` directory in the workflow repository root. This can be useful to make changes that affect task execution across all environments with a single change.
```